cpp/ql/lib/semmle/code/cpp/security/BufferWrite.qll

```diff
@@ -65,19 +65,7 @@ abstract class BufferWrite extends Expr {
    */
   int getExplicitLimit() { none() }
 
-  /**
-   * Gets an upper bound to the amount of data that's being written (if one
-   * can be found).
-   */
-  int getMaxData() { none() }
-
-  /**
-   * Gets an upper bound to the amount of data that's being written (if one
-   * can be found), except that float to string conversions are assumed to be
-   * much smaller (8 bytes) than their true maximum length.  This can be
-   * helpful in determining the cause of a buffer overflow issue.
-   */
-  int getMaxDataLimited() { result = this.getMaxData() }
+  int getMaxDataEstimate(TFormattedSizeEstimateKind strategy) { none() }
```

Comment:
`getMaxDataLimited()` had a default behaviour `result = this.getMaxData()`, which is used in some of the subclasses.